### PR TITLE
23 add black mypy checks

### LINKED ISSUE #23: Add black and mypy checks in Github Worflow
At the moment, black and mypy are added in dev packages but not used in the github workflow.
Should we automatically do the check ? It will require code development since it doesn't pass the check locally.

### PR DESCRIPTION
**Pull request by Wauplin**
Close #23 .
Add code quality check with Flake8, Black and Mypy tools. It required to change a lot of lines in the project. I think it's relevant to look more closely to what I've done even if the build checks pass. 

Also I have a comment on how to configure Flake8 correctly. See comment below.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -47,9 +47,15 @@ jobs:
...
 
-      # step by step
-      # - name: Code Quality Black
-      #   run: poetry run black . --check
+      # Code quality
+      - name: Code Quality Black
+        run: poetry run black . --check
+
+      - name: Code quality Flake8
+        run: poetry run flake8 . --ignore=E203,E266,E501,W503,W504,E741 --exclude=.git,__pycache__,.venv,.downloads,fhirzeug/generators/python/,fhirzeug/generators/python_pydantic/templates/
```
Comment (Wauplin) [inline .github/workflows/main.yml]: @julian-r : I can see that putting flake8 configuration inside the github workflow file is ugly but I did not find a better way of doing it yet. Flake8 project doesn't seem to support `pyproject.toml` config file and debate is raging on the subject : https://gitlab.com/pycqa/flake8/-/issues/428 . I did not want to include a `setup.cfg` file only for flake8 support so I left it here. 
Do you have any better solution to suggest ?

**Comment by Wauplin** [comment]
Code coverage :rage:

File: .github/workflows/main.yml (hunk 1 above)
Comment (julian-r) [reply .github/workflows/main.yml]: for me a [`.flake8`](https://flake8.pycqa.org/en/2.5.5/config.html#per-project) file would be fine.

File: tests/pydantic/test_basic_types_pyd.py [hunk 2]
```diff
@@ -45,6 +44,8 @@ def test_fhirdate():
     model = ExampleModel(date="1973-06")
     model = ExampleModel(date="1905-08-23")
 
+    del model
```
Comment (julian-r) [inline tests/pydantic/test_basic_types_pyd.py]: Why do we need this delete?

File: fhirzeug/fhirspec.py [hunk 3]
```diff
@@ -1194,8 +1200,5 @@ def __init__(self, mapping_arr):
         pass
 
 
-def _is_string(element):
-    isstr = isinstance(element, str)
-    if not isstr and sys.version_info[0] < 3:  # Python 2.x has 'str' and 'unicode'
-        isstr = isinstance(element, basestring)
-    return isstr
+def _is_string(element) -> bool:
```
Comment (julian-r) [inline fhirzeug/fhirspec.py]: actually we could also remove the function completely and replace the `_is_string` occurrences in the code....

**Review by julian-r — CHANGES_REQUESTED** [review]

File: tests/pydantic/test_basic_types_pyd.py (hunk 2 above)
Comment (Wauplin) [reply tests/pydantic/test_basic_types_pyd.py]: Hmm, not necessary indeed. The delete is purely artificial to use the `model` variable at least once (`F841 local variable 'model' is assigned to but never used`)

Comment (julian-r) [reply tests/pydantic/test_basic_types_pyd.py]: then I would rather disable the qa here than to change something semantically

**Comment by Wauplin** [comment]
@julian-r : I made the requested changes. Tests now pass successfully and coverage is valid. 
On my side, it's ready to be reviewed and merged.

**Review by julian-r — APPROVED** [review]

**Pull request merged**